server/aesdsocket.c

```diff
@@ -0,0 +1,200 @@
+		exit(EXIT_FAILURE);
+	}
+	//Get server address
+	hints.ai_flags=AI_PASSIVE; 
+	if(getaddrinfo(NULL,PORT_NUMBER,&hints,&servinfo) !=0) {
+		syslog(LOG_ERR, "Unable to get the server's address\n");
+		exit(EXIT_FAILURE);
+	}
+	
+	//Bind the socket 
+	if(bind(sock_fd,servinfo->ai_addr,sizeof(struct sockaddr)) == -1) {
+		syslog(LOG_ERR, "Unable to Bind\n");
+		freeaddrinfo(servinfo); 			//Freeing the memory created by socket address before exiting 
+		exit(EXIT_FAILURE);
+	}
+	freeaddrinfo(servinfo); 				//Freeing the memory created by socket address
+	
+	file_fd=open(PATH_TO_FILE, O_WRONLY|O_CREAT|O_TRUNC, S_IRWXU|S_IRGRP|S_IROTH); 	//Creating a new file with 744 octal value
+	if (file_fd==-1) {
+		syslog(LOG_ERR, "Unable to create the file\n");
+		exit(EXIT_FAILURE);
+	}
+	close(file_fd);
+
+	//Looping for all the connections from Client
+	while(1)
+	{
+		if(listen(sock_fd,MAX_PENDING_CONNECTIONS) == -1) {		//Will start rejecting after 32 pending connections 
+			syslog(LOG_ERR, "Unable to Listen to the Clients\n");
+			exit(EXIT_FAILURE);
+		}
+		conn_fd = accept(sock_fd,(struct sockaddr *)&clt_addr,&address_len);
+		if(conn_fd==-1) {
+			syslog(LOG_ERR, "Unable to Accept the Connection\n");
+			exit(EXIT_FAILURE);
+		}
+		syslog(LOG_INFO,"Accepts connection from %s",inet_ntoa(clt_addr.sin_addr));
+		printf("Accepts connection from %s\n",inet_ntoa(clt_addr.sin_addr));
+		h_dynamic_buff = (char*)malloc(sizeof(char) * BUFFER_SIZE); //Allocate size equal to buffer size
+		if(h_dynamic_buff==NULL) {
+			syslog(LOG_ERR, "Malloc Failure.Unable to Allocate Memory\n");
+			exit(EXIT_FAILURE);
+		}
+		memset(h_dynamic_buff,0,BUFFER_SIZE); //Reset the temporary buffer
+		packet_received=false;
+		while(!packet_received)
+		{
+		rec_status=recv(conn_fd,input_buffer,100,0); //Receive data packets from client
+		if(rec_status==-1)
+		{
+			syslog(LOG_ERR, "Error in reception of data packets from client");
+			exit(EXIT_FAILURE);
+		}
+		for(i=0;i<BUFFER_SIZE;i++)
+		{
+			if(input_buffer[i]=='\n')
+			{
+				i++;
+				packet_received=true;
+				break;
+			}	
+			
+		}
+		packet_bytes+=i;
+		h_dynamic_buff=(char *)realloc(h_dynamic_buff,packet_bytes);
+		if(h_dynamic_buff==NULL)
+		{
+			syslog(LOG_ERR, "Reallocation of memory failed");
+			exit(EXIT_FAILURE);
+		}
+		memcpy(h_dynamic_buff+packet_bytes-i,input_buffer,i); 
+		memset(input_buffer,0,BUFFER_SIZE); 
+		}
+
+		file_fd = open(PATH_TO_FILE,O_APPEND | O_WRONLY); 		//Open the file to write from current position
+		if(file_fd==-1) {
+			printf("Unable to open in write mode\n");
+			syslog(LOG_ERR, "Unable to open the File\n");
+			exit(EXIT_FAILURE);
+		}
+		if(write(file_fd,h_dynamic_buff,packet_bytes) !=packet_bytes) {
+			printf("Unable to write\n");
+			syslog(LOG_ERR, "All Bytes not written to the file\n");
+			exit(EXIT_FAILURE);
+		}
+		close(file_fd);
+
+		total_packets += packet_bytes; 			
+		char send_buffer[total_packets]; 			        //Storing contents to send
```

Comment:
@harshberiwal what are you assuming about the size of the stack and the total number of bytes sent here?